### DISCUSSION TITLE
supporting both onClickConnectStart/End and onConnectStart/End

### DISCUSSION THREAD

**Discussion by hmaiermack** [xyflow/xyflow#3006] [Q&A] [answered]
Hi, running into some issues trying to support creating a connection via click at the same time as supporting the default drag behavior. 

Default behavior always works, regardless of if I'm passing onClickConnect functions to the react flow instance. However, the onClickConnect behavior only works if I don't provide onConnectStart/End to the react flow instance. Is there a way to support both behaviors at the same time/ am I doing something wrong?

**Comment by bcakmakoglu** [comment]
Mh... I tried reproducing the issue in [this sandbox](https://codesandbox.io/s/unruffled-smoke-r2yqw3?file=/src/Flow.tsx) passing some basic handlers and didn't encounter any problems using drag-connect and click-connect together.

**Reply by hmaiermack** [reply]
Figured out that it's because I'm conditionally rendering the handle to provide an easy connect UX for dragging, which lead to the onClick not having the updated handle id. You can see the behavior in [this sandbox](https://codesandbox.io/s/objective-sinoussi-k277ju), I wasn't getting the handle id = null error in my local environment for some reason which would have made it much easier to figure out. I got around the issue by altering the z index of the handle to sit on top of the node when there was a click interaction. I don't know if this is a bug or if I'm just using the library weirdly :3

**Reply by bcakmakoglu** [reply]
[You might want to read this section of the docs.](https://reactflow.dev/docs/api/nodes/handle/#dynamic-handles)

**Reply by bcakmakoglu** [reply, marked as answer]
Quick fix for the issue would be this

```tsx
const CustomNode = ({
  id,
  data,
  isConnectable,
  targetPosition = Position.Top
}: NodeProps) => {
  const { isClickInteraction } = useNodeContext();
  const updateNodeInternals = useUpdateNodeInternals();

  useEffect(() => {
    updateNodeInternals(id);
  }, [isClickInteraction, updateNodeInternals, id]);
```

[Here's the sandbox with the fix.](https://codesandbox.io/s/flamboyant-snow-q7kez9?file=/src/CustomNode.tsx:362-520)

**Reply by hmaiermack** [reply]
Got it, I thought I might be missing something obvious, thanks for the help. Just to clarify, updateNodeInternals is only able to update one node, or can you pass it an array of ids?

**Reply by bcakmakoglu** [reply]
It only accepts a single id so you'd have to loop through the array and call it for each node :)

**Reply by bcakmakoglu** [reply]
I opened a PR to potentially support an array of ids as well #3007
